apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -181,33 +209,23 @@ export default class AmazonFutureEngineerEligibility extends React.Component {
   };
 
   loadCompletionPage = () => {
-    // Dedupe with loadConfirmationPage -- probably belongs in onContinue
-    this.saveToSessionStorage();
-
-    // Do API calls here.
-    if (!this.state.formData.submitted) {
-      this.submitToAFE();
+    let sessionEligibilityData = this.getSessionEligibilityData();
+
+    if (!sessionEligibilityData.submitted) {
+      this.submitToAFE().then(() => {
+        sessionStorage.setItem(
+          sessionStorageKey,
+          JSON.stringify({...sessionEligibilityData, submitted: true})
```

Comment:
what happens if the submit fails?

Reply:
To be added, @islemaster and I got just got the API hooked up in a subsequent PR, so I will add something to handle failure.